homeassistant/components/zeroconf/__init__.py

```diff
@@ -98,16 +98,43 @@
 
 @dataclass(slots=True)
 class ZeroconfServiceInfo(BaseServiceInfo):
-    """Prepared info from mDNS entries."""
+    """Prepared info from mDNS entries.
 
-    host: str
-    addresses: list[str]
+    The ip_address is the most recently updated address
+    that is not a link local or unspecified address.
+
+    The ip_addresses are all addresses in order of most
+    recently updated to least recently updated.
+
+    The host is the string representation of the ip_address.
+
+    The addresses are the string representations of the
+    ip_addresses.
+
+    It is recommended to use the ip_address to determine
+    the address to connect to as it will be the most
+    recently updated address that is not a link local
+    or unspecified address.
```

Comment:
These are a combination of the guarantees that python-zeroconf makes and our code to process ServiceInfo